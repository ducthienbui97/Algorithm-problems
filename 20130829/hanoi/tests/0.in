90
1
CB BA AC CA BC AB
1
CB BC BA CA AB AC
1
AC CA BC CB BA AB
2
AB CA BC BA AC CB
2
AC AB CB BC BA CA
2
AB BA AC CA BC CB
3
AC BA AB BC CB CA
3
CB BC AB CA AC BA
3
BA CB BC AB AC CA
4
BC AB BA CA CB AC
4
BC CB AC CA BA AB
4
CB BA AB BC AC CA
5
CB BA CA BC AC AB
5
BC AC CB CA BA AB
5
BC CA BA AC CB AB
6
CB BA AC CA AB BC
6
CB AC AB CA BC BA
6
BA AB CA BC AC CB
7
BC CA CB AB AC BA
7
AB CB CA AC BC BA
7
CA AC BC CB BA AB
8
CB AC CA BA AB BC
8
CB CA AC BC BA AB
8
AB AC CB BA BC CA
9
AB AC BC CA BA CB
9
AC CB BC AB CA BA
9
AB CA AC CB BA BC
10
BA AC CB CA BC AB
10
AC CB BC BA AB CA
10
CA BA BC AC AB CB
11
BC CA CB AB BA AC
11
BC CB BA CA AB AC
11
CA BA BC AC CB AB
12
CB BA AC AB CA BC
12
CB BC BA AB AC CA
12
BA CB AB BC AC CA
13
CB BA CA BC AC AB
13
CB BC AC AB CA BA
13
CA AC BA BC AB CB
14
AB BC BA AC CA CB
14
BC CB CA AB BA AC
14
CA BA BC AC AB CB
15
CA AB CB AC BC BA
15
CB BC CA BA AC AB
15
BA BC CA AC CB AB
16
BA AC BC CB CA AB
16
AC CB CA BC AB BA
16
CA AB BA CB BC AC
17
BA CB AC AB CA BC
17
AB CB BC BA AC CA
17
CA AC BA CB AB BC
18
CA CB BC BA AB AC
18
CB CA BC BA AB AC
18
AC CA CB AB BA BC
19
CA AB AC BC CB BA
19
CB AC AB BC BA CA
19
CA BA AC BC AB CB
20
BC BA CA CB AB AC
20
AC AB BC BA CB CA
20
BA AB BC CB AC CA
21
AB AC CA BC CB BA
21
CB AC CA BC BA AB
21
CA BA BC AB AC CB
22
CA AB BC BA AC CB
22
CB BC CA AB AC BA
22
CA BA CB AB BC AC
23
AB AC CA BC CB BA
23
CB BC BA CA AC AB
23
BA CA AB BC CB AC
24
CA CB BC BA AB AC
24
BC BA CB CA AB AC
24
CA CB BC AC AB BA
25
BC BA CA CB AB AC
25
CB CA BC AB BA AC
25
AB BA BC AC CA CB
26
BC CA CB AB BA AC
26
CB BC BA CA AC AB
26
CA BA BC CB AB AC
27
CA AB CB BC BA AC
27
AB CB AC BC BA CA
27
CA BA CB AB AC BC
28
AC BA CB AB BC CA
28
BC BA CB AB AC CA
28
BA AB CA CB AC BC
29
CA CB AB BC AC BA
29
CB AB BC BA AC CA
29
AB BA CB CA AC BC
30
BA CB CA AC AB BC
30
AC BC BA AB CB CA
30
BA AC BC CA CB AB
