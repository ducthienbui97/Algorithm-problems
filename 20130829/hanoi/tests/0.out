1
1
1
3
3
5
7
9
17
15
27
53
31
81
161
63
243
485
127
729
1457
255
2187
4373
511
6561
13121
1023
19683
39365
2047
59049
118097
4095
177147
354293
8191
531441
1062881
16383
1594323
3188645
32767
4782969
9565937
65535
14348907
28697813
131071
43046721
86093441
262143
129140163
258280325
524287
387420489
774840977
1048575
1162261467
2324522933
2097151
3486784401
6973568801
4194303
10460353203
20920706405
8388607
31381059609
62762119217
16777215
94143178827
188286357653
33554431
282429536481
564859072961
67108863
847288609443
1694577218885
134217727
2541865828329
5083731656657
268435455
7625597484987
15251194969973
536870911
22876792454961
45753584909921
1073741823
68630377364883
137260754729765
